lib/system/external.toit

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2024 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import io
+import rpc
+
+class ExternalMessageHandler_ implements SystemMessageHandler_:
+  static TYPE ::= SYSTEM-EXTERNAL-NOTIFICATION_
+  handler/Lambda := ?
+
+  constructor .handler:
+    set-system-message-handler_ TYPE this
+
+  close -> none:
+    clear-system-message-handler_ TYPE
+
+  on-message type/int gid/int pid/int argument -> none:
+    handler.call argument
+
+externals_ := {:}
+
+class External:
+  pid/int
+  id/string
+  external-message-handler_/ExternalMessageHandler_? := null
+  is-closed_/bool := false
+
+  constructor.private_ .pid .id:
+
+  static get id/string -> External?:
+    result := externals_.get id
+    if result: return result
+    pid := pid-for-external-id_ id
+    if pid == -1: return null
+    result = External.private_ pid id
+    externals_[id] = result
```

Comment:
Not sure I understand the design here. You're caching instances of `External`, but they come with a close. That feels shaky. Can you ever reopen them?